internal/framework/status/updater.go

```diff
@@ -250,6 +253,21 @@ func (upd *UpdaterImpl) writeStatuses(
 	}
 }
 
+// UpdateAddresses is called when the Gateway Status needs its addresses updated.
+func (upd *UpdaterImpl) UpdateAddresses(ctx context.Context, addresses []v1beta1.GatewayStatusAddress) {
+	upd.lock.Lock()
+	for name, status := range upd.lastStatuses.gatewayAPI.GatewayStatuses {
+		if status.Ignored {
+			continue
+		}
+		status.Addresses = addresses
+		upd.lastStatuses.gatewayAPI.GatewayStatuses[name] = status
+	}
+	upd.lock.Unlock()
+
+	upd.Update(ctx, upd.lastStatuses.gatewayAPI)
```

Comment:
I think there is still a data race here. Reading from `upd.lastStatuses.gatewayAPI` can happen at the same time while `upd.lastStatuses.gatewayAPI` is updated in another goroutine

Reply:
True, I actually don't think I need to call `Update()`, I can call `updateGatewayAPI` and keep the lock.